File: cli/src/theme.ts
import chalk, { type ChalkInstance } from 'chalk'

/** Skin color map - matches lib/skins.ts on the server */
const SKIN_COLORS: Record<string, string> = {
  monochrome: '#FFFFFF',
  red: '#EF4444',
  blue: '#3B82F6',
  navy: '#1E3A8A',
  indigo: '#6366F1',
  purple: '#A855F7',
  teal: '#14B8A6',
  green: '#22C55E',
  amber: '#F59E0B',
  rose: '#F43F5E',
}

let accentColor = '#FFFFFF'

/** Set the accent color from the user's skin preference */
export function setAccent(skinId: string, customHex?: string | null): void {
  if (skinId === 'hex' && customHex) {
    accentColor = customHex
  } else {
    accentColor = SKIN_COLORS[skinId] ?? '#FFFFFF'
  }
}

/** Returns a chalk instance using the user's accent color */
export function accent(text: string): string {
  return chalk.hex(accentColor)(text)
}

/** Bold accent */
export function accentBold(text: string): string {
  return chalk.hex(accentColor).bold(text)
}

/** Dimmed text */
export function dim(text: string): string {
  return chalk.dim(text)
}

/** Upvote color (green) */
export function up(text: string): string {
  return chalk.green(text)
}

/** Downvote color (red) */
export function down(text: string): string {
  return chalk.red(text)
}

/** Board tag */
export function board(text: string): string {
  return chalk.hex(accentColor)(`#${text}`)
}

/** Separator line */
export function separator(): string {
  return chalk.dim('─'.repeat(60))
}

File: cli-rust/src/theme.rs
use colored::Colorize;

/// Skin color map matching lib/skins.ts
fn skin_hex(id: &str) -> &str {
    match id {
        "red" => "#EF4444",
        "blue" => "#3B82F6",
        "navy" => "#1E3A8A",
        "indigo" => "#6366F1",
        "purple" => "#A855F7",
        "teal" => "#14B8A6",
        "green" => "#22C55E",
        "amber" => "#F59E0B",
        "rose" => "#F43F5E",
        _ => "#FFFFFF",
    }
}

fn parse_hex(hex: &str) -> (u8, u8, u8) {
    let hex = hex.trim_start_matches('#');
    let r = u8::from_str_radix(&hex[0..2], 16).unwrap_or(255);
    let g = u8::from_str_radix(&hex[2..4], 16).unwrap_or(255);
    let b = u8::from_str_radix(&hex[4..6], 16).unwrap_or(255);
    (r, g, b)
}

pub fn get_accent(skin_id: &str, custom_hex: Option<&str>) -> (u8, u8, u8) {
    if skin_id == "hex" {
        if let Some(hex) = custom_hex {
            return parse_hex(hex);
        }
    }
    parse_hex(skin_hex(skin_id))
}

pub fn accent(text: &str, color: (u8, u8, u8)) -> String {
    text.truecolor(color.0, color.1, color.2).to_string()
}

pub fn accent_bold(text: &str, color: (u8, u8, u8)) -> String {
    text.truecolor(color.0, color.1, color.2).bold().to_string()
}

pub fn dim(text: &str) -> String {
    text.dimmed().to_string()
}

pub fn green(text: &str) -> String {
    text.green().to_string()
}

pub fn red(text: &str) -> String {
    text.red().to_string()
}

pub fn board(name: &str, color: (u8, u8, u8)) -> String {
    format!("#{}", name).truecolor(color.0, color.1, color.2).to_string()
}

pub fn separator() -> String {
    "─".repeat(60).dimmed().to_string()
}
